### PR TITLE
fix(curriculum): rework Project Euler 92

### PR DESCRIPTION
**Pull request by gikf**
- Reworks challenge to use argument in function call.
- Adds more tests.
- Adds solution.
- Positions block with examples evenly between paragraphs.
- Related to #40896.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-92-square-digit-chains.md [hunk 1]
```diff
@@ -15,41 +15,141 @@ For example,
 <div style='margin-left: 4em;'>
   44 → 32 → 13 → 10 → <strong>1</strong> → <strong>1</strong><br>
   85 → <strong>89</strong> → 145 → 42 → 20 → 4 → 16 → 37 → 58 → <strong>89</strong>
-</div>
+</div><br>
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-92-square-digit-chains.md:18]: Can this bit be better represented with KaTeX?

Comment (gikf) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-92-square-digit-chains.md:18]: Yes! Might be even nicer than expected:
image
What do you think?

```suggestion
$$\begin{align}
  & 44 → 32 → 13 → 10 → \boldsymbol{1} → \boldsymbol{1}\\\\
  & 85 → \boldsymbol{89} → 145 → 42 → 20 → 4 → 16 → 37 → 58 → \boldsymbol{89}\\\\
\end{align}$$
```

Comment (nhcarrigan) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-92-square-digit-chains.md:18]: Yep I think that's the way to go.

**Review by Sembauke — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]

**Comment by moT01** [comment]
I added this solution to the forum guide article

**Pull request merged**